javasource/oql/implementation/OQL.java

```diff
@@ -96,6 +95,14 @@ public static List<IMendixObject> executeOQL(IContext context, String statement,
 			IMendixObject targetObj = Core.instantiate(context, returnEntity);
 			for (int i = 0; i < tableSchema.getColumnCount(); i++) {
 				IDataColumnSchema columnSchema = tableSchema.getColumnSchema(i);
+
+				if (columnSchema.getName().matches(".+\\.ID$")) {

```

Comment:
My understanding is that the idea here is only to skip IDs if they do not match any associated entities. In that sense, it would be more precise to have this check every time we call `getAssociation(targetObj, columnSchema)` or embed it in that function.

Reply:
ID is a reserved word. You cannot have any attribute (or association) with that name, so to me it makes sense to skip the entire column because there's only one possibility here.